src/Native/Testable/Navigation.js

```diff
@@ -0,0 +1,110 @@
+if (typeof _elm_lang$navigation$Native_Navigation === 'undefined') { // eslint-disable-line camelcase
+  throw new Error('Native.Testable.Navigation was loaded before _elm_lang$navigation$Native_Navigation: this shouldn\'t happen because Testable.Navigation imports Navigation. Please report this at https://github.com/avh4/elm-testable/issues')
+}
+
+var original_getLocation = _elm_lang$navigation$Native_Navigation.getLocation;
+_elm_lang$navigation$Native_Navigation.getLocation = function () {
+  if (typeof document !== 'undefined' && typeof document.location !== 'undefined') {
+    return original_getLocation()
+  }
+
+  return {
+    href: 'https://elm.testable/',
+    host: 'elm.testable',
+    hostname: 'elm.testable',
+    protocol: 'https:',
+    origin: 'https://elm.testable',
+    port_: '',
+    pathname: '/',
+    search: '',
+    hash: '',
+    username: '',
+    password: ''
+  }
+}
+
+var mapNavigation = function(ctor) {
+  return function (data) {
+      return {
+        ctor: 'Navigation_NativeNavigation',
+        _0: {
+          ctor: ctor,
+          _0: data
+        },
+        _1: function (data) { return { ctor: 'Success', _0: data } }
+      }
+  }
+}
+
+_elm_lang$navigation$Navigation$pushState = setItUp(
+  _elm_lang$navigation$Navigation$pushState,
+  mapNavigation('New')
+)
+
+_elm_lang$navigation$Navigation$replaceState = setItUp(
+  _elm_lang$navigation$Navigation$replaceState,
+  mapNavigation('Modify')
+)
+
+_elm_lang$navigation$Navigation$go = setItUp(
+  _elm_lang$navigation$Navigation$go,
+  mapNavigation('Jump')
+)
+
+_elm_lang$navigation$Navigation$setLocation = setItUp(
+  _elm_lang$navigation$Navigation$setLocation,
+  mapNavigation('Visit')
```

Comment:
I think it would be preferable to have these ctors named after the functions they're intercepting, so `PushState/ReplaceState/Go/SetLocation` instead of `New/Modify/Jump/Visit`